File: frontend/src/types/page.ts
/**
 * Page Types - 页面/文档类型定义
 */

export interface Page {
  id: number
  title: string
  content: string
  content_json?: string | object
  published: boolean
  archived_at?: string
  sequence: number
  parent_id?: number
  depth: number
  locked_by_id?: number
  locked_at?: string
  locked_by_name?: string
  project_id: number
  workspace_id: number
  created_by_id?: number
  updated_by_id?: number
  created_at: string
  updated_at: string
  children?: Page[]
}

export interface PageCreate {
  title: string
  content?: string
  content_json?: string
  parent_id?: number
  sequence?: number
}

export interface PageUpdate {
  title?: string
  content?: string
  content_json?: string
  published?: boolean
  sequence?: number
}

export interface PageMove {
  parent_id?: number | null
  sequence: number
}

export interface PageVersion {
  id: number
  page_id: number
  title: string
  content: string
  content_json?: string
  version_number: number
  change_summary?: string
  created_at: string
  created_by_id?: number
  created_by_name?: string
}

export interface PageTemplate {
  id: number
  name: string
  description: string
  content: string
  content_json?: string
  is_default: boolean
  workspace_id: number
  project_id?: number
  created_at: string
  updated_at: string
}

export interface PageTemplateCreate {
  name: string
  description?: string
  content?: string
  content_json?: string
  is_default?: boolean
  project_id?: number
}

export interface PageTemplateUpdate {
  name?: string
  description?: string
  content?: string
  content_json?: string
  is_default?: boolean
}

File: backend/internal/dto/response/page.go
package response

import "time"

// PageResponse is the API response for a page.
type PageResponse struct {
	ID          uint64     `json:"id"`
	Title       string     `json:"title"`
	Content     string     `json:"content"`
	ContentJSON *string    `json:"content_json"`
	Published   bool       `json:"published"`
	ArchivedAt  *time.Time `json:"archived_at"`
	Sequence    int        `json:"sequence"`
	ParentID    *uint64    `json:"parent_id"`
	Depth       int        `json:"depth"`

	// Locking
	LockedByID   *uint64    `json:"locked_by_id"`
	LockedAt     *time.Time `json:"locked_at"`
	LockedByName string     `json:"locked_by_name,omitempty"`

	ProjectID   uint64 `json:"project_id"`
	WorkspaceID uint64 `json:"workspace_id"`

	CreatedByID *uint64   `json:"created_by_id"`
	UpdatedByID *uint64   `json:"updated_by_id"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`

	Children []PageResponse `json:"children,omitempty"`
}
